src/transform/CollapseTable.js

```diff
@@ -3,34 +3,154 @@ import Polyfill from './Polyfill'
 import elementUtilities from './ElementUtilities'
 
 const SECTION_TOGGLED_EVENT_TYPE = 'section-toggled'
+const ELEMENT_NODE = 1
+const TEXT_NODE = 3
+
+/**
+ * Determine if we want to extract text from this header.
+ * @param {!Element} header
+ * @return {!boolean}
+ */
+const isHeaderEligible =
+  header => header.childNodes && Polyfill.querySelectorAll(header, 'a').length < 3
+
+/**
+ * Determine eligibility of extracted text.
+ * @param {?string} headerText
+ * @return {!boolean}
+ */
+const isHeaderTextEligible = headerText => !(
+  !headerText || headerText.trim().length === 0 || headerText.replace(/[\s0-9]/g, '').length === 0
+)
+
+/**
+ * Extracts first word from string. Returns null if for any reason it is unable to do so.
+ * @param  {!string} string
+ * @return {?string}
+ */
+const firstWordFromString = string => {
+  // 'If the global flag (g) is not set, Element zero of the array contains the entire match,
+  // while elements 1 through n contain any submatches.'
+  const matches = string.match(/\b(\w+)\b/) // Only need first match so not using 'g' option.
+  if (!matches || matches.length === 0) {
```

Comment:
I got rid of the `length` check - I should have caught from the docs that if a match is found you get an array.

Also made it return `undefined` instead of `null`. I did keep it explicit though.